internals/overlord/state/identities.go

```diff
@@ -130,10 +187,11 @@ func (s *State) AddIdentities(identities map[string]*Identity) error {
 		if _, ok := s.identities[name]; ok {
 			existing = append(existing, name)
 		}
-		err := identity.validate()
+		err := identity.validate(name)
 		if err != nil {
 			return fmt.Errorf("identity %q invalid: %w", name, err)
 		}
+		identity.Name = name
```

Comment:
We should remove this here, as we still check for further errors below (and it's set below anyway).